pkg/templateservicebroker/servicebroker/provision.go

```diff
@@ -268,8 +268,15 @@ func (b *Broker) ensureBrokerTemplateInstance(u user.Info, namespace, instanceID
 		}
 
 		existingBrokerTemplateInstance, err := b.templateclient.BrokerTemplateInstances().Get(brokerTemplateInstance.Name, metav1.GetOptions{})
-		if err == nil && reflect.DeepEqual(brokerTemplateInstance.Spec, existingBrokerTemplateInstance.Spec) {
-			return existingBrokerTemplateInstance, nil
+		if err == nil &&
+			brokerTemplateInstance.Spec.TemplateInstance.Kind == existingBrokerTemplateInstance.Spec.TemplateInstance.Kind &&
+			brokerTemplateInstance.Spec.TemplateInstance.Namespace == existingBrokerTemplateInstance.Spec.TemplateInstance.Namespace &&
+			brokerTemplateInstance.Spec.TemplateInstance.Name == existingBrokerTemplateInstance.Spec.TemplateInstance.Name &&
+			brokerTemplateInstance.Spec.Secret.Kind == existingBrokerTemplateInstance.Spec.Secret.Kind &&
+			brokerTemplateInstance.Spec.Secret.Namespace == existingBrokerTemplateInstance.Spec.Secret.Namespace &&
+			brokerTemplateInstance.Spec.Secret.Name == existingBrokerTemplateInstance.Spec.Secret.Name {
+			glog.V(4).Infof("Provision requested for instance %s, but job is already in progress", instanceID)
+			return nil, api.NewResponse(http.StatusOK, api.ProvisionResponse{Operation: api.OperationProvisioning}, nil)
```

Comment:
hm, I see one problem w/ this that I did not think of before:  With this check, a user will never be able to provision two instances of a given service offering.

Reply:
I will update to have the ensureBrokerTemplateInstance() return the existingTemplateInstance as it did before.